addons/account/models/partner.py

```diff
@@ -565,6 +569,24 @@ def _compute_show_credit_limit(self):
         for partner in self:
             partner.show_credit_limit = self.env.company.account_use_credit_limit
 
+    api.depends('credit', 'total_invoiced')
+    def _compute_days_sales_outstanding(self):
+        for partner in self:
+            all_partner_ids = self.with_context(active_test=False).search(
+                [('id', 'child_of', partner.id)]).ids
+            first_invoice = self.env['account.move'].sudo().search(
+                domain=[('partner_id', 'in', all_partner_ids),
+                        ('move_type', '=', 'out_invoice'),
+                        ('state', 'not in', ['cancel', 'draft'])],
+                order='invoice_date asc',
+                limit=1)
```

Comment:
doing a search inside of a loop is not efficient, you'd go ahead of performance issues.